--- ddclient-3.7.1/ddclient.orig	2007-04-28 14:31:23.000000000 +0100
+++ ddclient-3.7.1/ddclient	2007-04-28 14:31:33.000000000 +0100
@@ -21,7 +21,7 @@
 $program  =~ s/d$//;
 my $now       = time;
 my $hostname  = hostname();
-my $etc       = ($program =~ /test/i) ? './'   : '/etc/ddclient/';
+my $etc       = ($program =~ /test/i) ? './'   : '/etc/';
 my $cachedir  = ($program =~ /test/i) ? './'   : '/var/cache/ddclient/';
 my $savedir   = ($program =~ /test/i) ? 'URL/' : '/tmp/';
 my $msgs      = '';
