Index: ddclient-3.7.0/ddclient
===================================================================
--- ddclient-3.7.0.orig/ddclient	2006-12-01 23:22:36.919929347 +0100
+++ ddclient-3.7.0/ddclient	2006-12-01 23:25:15.960666685 +0100
@@ -966,6 +966,8 @@
 sub init_config {
     %opt = %saved_opt;
 
+    $opt{'daemon'}=interval($opt{'daemon'}) if defined $opt{'daemon'};
+
     ## 
     $opt{'quiet'}   = 0 if   opt('verbose');
 

