Index: ddclient-3.6.2/sample-etc_cron.d_ddclient
===================================================================
--- ddclient-3.6.2.orig/sample-etc_cron.d_ddclient	2002-01-04 00:56:34.000000000 +0100
+++ ddclient-3.6.2/sample-etc_cron.d_ddclient	2005-10-18 23:41:12.000000000 +0200
@@ -10,7 +10,7 @@
 ######################################################################
 ## force an update twice a month
 ##
-30 23 1,15 * *	root    /root/bin/ddclient -daemon=0 -syslog -quiet -force
+30 23 1,15 * *	root    /usr/sbin/ddclient -daemon=0 -syslog -quiet -force
 ######################################################################
 ## retry failed updates every hour (only if you are not using daemon-mode)
 ##
Index: ddclient-3.6.2/sample-etc_dhcpc_dhcpcd-eth0.exe
===================================================================
--- ddclient-3.6.2.orig/sample-etc_dhcpc_dhcpcd-eth0.exe	2005-10-18 23:40:22.000000000 +0200
+++ ddclient-3.6.2/sample-etc_dhcpc_dhcpcd-eth0.exe	2005-10-18 23:41:02.000000000 +0200
@@ -15,6 +15,6 @@
 192.168.*)	;;
 *)
 	logger -t dhcpcd IP address changed to $1
-	/root/bin/ddclient -daemon=0 -syslog -use=ip -ip=$1 >/dev/null 2>&1
+	/usr/sbin/ddclient -daemon=0 -syslog -use=ip -ip=$1 >/dev/null 2>&1
 	;;
 esac

