#
# patch added to revision 28
#
Index: ddclient
===================================================================
--- ddclient	(revision 27)
+++ ddclient	(working copy)
@@ -593,7 +593,7 @@
     usage("invalid argument '-use %s'; possible values are:\n%s", $opt{'use'}, join("\n",ip_strategies_usage()))
       unless exists $ip_strategies{lc opt('use')};
     
-    $daemon = define($opt{'daemon'}, $globals{'daemon'});
+    $daemon = $opt{'daemon'};
     $daemon = 0 if opt('force');
 
     ## obtain the IP address to use.
@@ -979,9 +979,13 @@
     $opt{'min-error-interval'} = max(interval(opt('min-error-interval')), interval(default('min-error-interval')));
 
     $opt{'timeout'}  = 0               if opt('timeout') < 0;
-    $opt{'daemon'}   = minimum('daemon')
-      if   define($opt{'daemon'},$globals{'daemon'},0) 
-	&& define($opt{'daemon'},$globals{'daemon'}) < minimum('daemon');
+
+    ## only set $opt{'daemon'} if it has been explicitly passed in
+    if (define($opt{'daemon'},$globals{'daemon'},0)) {
+        $opt{'daemon'} = interval(opt('daemon'));
+        $opt{'daemon'} = minimum('daemon')
+          if ($opt{'daemon'} < minimum('daemon'));
+    }
     
     ## define or modify host options specified on the command-line
     if (exists $opt{'options'} && defined $opt{'options'}) {
