Index: ddclient
===================================================================
--- ddclient	(revision 88)
+++ ddclient	(working copy)
@@ -1996,7 +1996,7 @@
     my $sub    = shift;
     my $update = 0;
 
-    if ($config{$host}{'login'} eq '') {
+    if (!($config{$host}{'login'})) {
 	warning("null login name specified for host %s.", $host);
 
     } elsif ($config{$host}{'password'} eq '') {
