Index: ddclient-3.7.0/ddclient
===================================================================
--- ddclient-3.7.0.orig/ddclient	2006-12-01 23:22:37.043910433 +0100
+++ ddclient-3.7.0/ddclient	2006-12-01 23:24:59.469182565 +0100
@@ -2231,7 +2231,7 @@
        '!yours'   => 'The hostname specified exists, but not under the username currently being used',
        '!donator' => 'The offline setting was set, when the user is not a donator',
        '!active'  => 'The hostname specified is in a Custom DNS domain which has not yet been activated.',
-       'abuse',   => 'The hostname specified is blocked for abuse; contact support@dyndns.org to be unblocked',
+       'abuse',   => 'The hostname specified is blocked for abuse; fill in the form at http://support.dyndns.org/abuse.php to be unblocked',
 
        'numhost'  => 'System error: Too many or too few hosts found. Contact support@dyndns.org',
        'dnserr'   => 'System error: DNS error encountered. Contact support@dyndns.org',

