Index: ddclient-3.7.0/ddclient
===================================================================
--- ddclient-3.7.0.orig/ddclient	2006-12-01 23:22:37.023913484 +0100
+++ ddclient-3.7.0/ddclient	2006-12-01 23:25:03.977494794 +0100
@@ -993,8 +993,8 @@
 
     $opt{'timeout'}  = 0               if opt('timeout') < 0;
     $opt{'daemon'}   = minimum('daemon')
-      if   define($opt{'daemon'},$globals{'daemon'},0) 
-	&& define($opt{'daemon'},$globals{'daemon'}) < minimum('daemon');
+      if   interval(define($opt{'daemon'},$globals{'daemon'},0))
+	&& interval(define($opt{'daemon'},$globals{'daemon'})) < minimum('daemon');
     
     ## define or modify host options specified on the command-line
     if (exists $opt{'options'} && defined $opt{'options'}) {

