Index: ddclient-3.7.0/sample-etc_ppp_ip-up.local
===================================================================
--- ddclient-3.7.0.orig/sample-etc_ppp_ip-up.local	2006-06-14 21:59:40.000000000 +0200
+++ ddclient-3.7.0/sample-etc_ppp_ip-up.local	2006-12-01 23:22:37.101901586 +0100
@@ -34,7 +34,7 @@
 		;;
 *)		(
 			sleep 5
-			ddclient -daemon=0 -syslog -use=if -if=$1 >/dev/null 2>&1
+			ddclient -daemon=0 -syslog -use=if -if=$IP >/dev/null 2>&1
 		) &
 		;;
 esac

