Index: ddclient-3.7.0/ddclient
===================================================================
--- ddclient-3.7.0.orig/ddclient	2006-12-01 23:22:37.062907535 +0100
+++ ddclient-3.7.0/ddclient	2006-12-01 23:24:53.543086628 +0100
@@ -357,7 +357,7 @@
 	'atime'               => setv(T_NUMBER, 0, 1, 0, 0,                   undef),
 	'status'              => setv(T_ANY,    0, 1, 0, '',                  undef),
 	'min-interval'        => setv(T_DELAY,  0, 0, 1, interval('30s'),     0),
-	'max-interval'        => setv(T_DELAY,  0, 0, 1, interval('15d'),     0),
+	'max-interval'        => setv(T_DELAY,  0, 0, 1, interval('30d'),     0),
 	'min-error-interval'  => setv(T_DELAY,  0, 0, 1, interval('5m'),      0),
 
 	'warned-min-interval'       => setv(T_ANY,    0, 1, 0, 0,             undef),

