Index: ddclient-3.7.0/ddclient
===================================================================
--- ddclient-3.7.0.orig/ddclient	2006-12-01 23:22:37.003916535 +0100
+++ ddclient-3.7.0/ddclient	2006-12-01 23:25:06.131166238 +0100
@@ -73,6 +73,11 @@
 				  'url'  => '/status.htm',
 				  'skip' => 'IP Address',
 			        },
+    'smc-barricade-alt'      => {
+				  'name' => 'SMC Barricade FW (alternate config)',
+				  'url'  => '/status.HTM',
+				  'skip' => 'WAN IP',
+			        },
     'smc-barricade-7401bra'  => {
 				  'name' => 'SMC Barricade 7401BRA FW',
 				  'url'  => '/admin/wan1.htm',

