--- ddclient-3.7.1/ddclient.orig	2007-04-30 00:40:49.000000000 +0100
+++ ddclient-3.7.1/ddclient	2007-04-30 00:40:59.000000000 +0100
@@ -54,7 +54,7 @@

 ## strategies for obtaining an ip address.
 my %builtinweb = (
-   'dyndns'       => { 'url' => 'http://checkip.dyndns.org/', 'skip' => 'Current IP Address', },
+   'dyndns'       => { 'url' => 'http://checkip.dyndns.com/', 'skip' => 'Current IP Address:', },
    'dnspark'      => { 'url' => 'http://ipdetect.dnspark.com/', 'skip' => 'Current Address:', },
 );
 my %builtinfw = (
@@ -342,7 +342,7 @@
 	'postscript'          => setv(T_POSTS, 0, 0, 1, '',                   undef),
     },
     'service-common-defaults'       => {
-	'server'	      => setv(T_FQDNP,  1, 0, 1, 'members.dyndns.org', undef),
+	'server'	      => setv(T_FQDNP,  1, 0, 1, 'members.dyndns.com', undef),
 	'login'               => setv(T_LOGIN,  1, 0, 1, '',                  undef),
 	'password'            => setv(T_PASSWD, 1, 0, 1, '',                  undef),
 	'host'                => setv(T_STRING, 1, 1, 1, '',                  undef),
@@ -1915,7 +1915,7 @@
 For example:
   protocol=hammernode1, \\
   login=my-hn-login, password=my-hn-password  myhost.hn.org
-  login=my-login, password=my-password  myhost.dyndns.org,my2nd.dyndns.org
+  login=my-login, password=my-password  myhost.dyndns.com,my2nd.dyndns.com

 specifies two host definitions.

@@ -1925,7 +1925,7 @@

 The second host definition will use the current default protocol
 ('dyndns2'), my-login and my-password to update the ip-address of
-myhost.dyndns.org and my2ndhost.dyndns.org.
+myhost.dyndns.com and my2ndhost.dyndns.com.

 The order of this sequence is significant because the values of any
 global variable definitions are bound to a host definition when the
@@ -2065,13 +2065,13 @@
 o 'dyndns1'

 The 'dyndns1' protocol is a deprecated protocol used by the free dynamic
-DNS service offered by www.dyndns.org. The 'dyndns2' should be used to
-update the www.dyndns.org service.  However, other services are also
+DNS service offered by www.dyndns.com. The 'dyndns2' should be used to
+update the www.dyndns.com service.  However, other services are also
 using this protocol so support is still provided by ${program}.

 Configuration variables applicable to the 'dyndns1' protocol are:
   protocol=dyndns1             ##
-  server=fqdn.of.service       ## defaults to members.dyndns.org
+  server=fqdn.of.service       ## defaults to members.dyndns.com
   backupmx=no|yes              ## indicates that this host is the primary MX for the domain.
   mx=any.host.domain           ## a host MX'ing for this host definition.
   wildcard=no|yes              ## add a DNS wildcard CNAME record that points to {host}
@@ -2082,16 +2082,16 @@
 Example ${program}.conf file entries:
   ## single host update
   protocol=dyndns1,                                         \\
-  login=my-dyndns.org-login,                                \\
-  password=my-dyndns.org-password                           \\
-  myhost.dyndns.org
+  login=my-dyndns.com-login,                                \\
+  password=my-dyndns.com-password                           \\
+  myhost.dyndns.com

   ## multiple host update with wildcard'ing mx, and backupmx
   protocol=dyndns1,                                         \\
-  login=my-dyndns.org-login,                                \\
-  password=my-dyndns.org-password,                          \\
+  login=my-dyndns.com-login,                                \\
+  password=my-dyndns.com-password,                          \\
   mx=a.host.willing.to.mx.for.me,backupmx=yes,wildcard=yes  \\
-  myhost.dyndns.org,my2ndhost.dyndns.org
+  myhost.dyndns.com,my2ndhost.dyndns.com
 EoEXAMPLE
 }
 ######################################################################
@@ -2177,13 +2177,13 @@
 o 'dyndns2'

 The 'dyndns2' protocol is a newer low-bandwidth protocol used by a
-free dynamic DNS service offered by www.dyndns.org.  It supports
+free dynamic DNS service offered by www.dyndns.com.  It supports
 features of the older 'dyndns1' in addition to others.  [These will be
 supported in a future version of ${program}.]

 Configuration variables applicable to the 'dyndns2' protocol are:
   protocol=dyndns2             ##
-  server=fqdn.of.service       ## defaults to members.dyndns.org
+  server=fqdn.of.service       ## defaults to members.dyndns.com
   backupmx=no|yes              ## indicates that this host is the primary MX for the domain.
   static=no|yes                ## indicates that this host has a static IP address.
   custom=no|yes                ## indicates that this host is a 'custom' top-level domain name.
@@ -2196,21 +2196,21 @@
 Example ${program}.conf file entries:
   ## single host update
   protocol=dyndns2,                                         \\
-  login=my-dyndns.org-login,                                \\
-  password=my-dyndns.org-password                           \\
-  myhost.dyndns.org
+  login=my-dyndns.com-login,                                \\
+  password=my-dyndns.com-password                           \\
+  myhost.dyndns.com

   ## multiple host update with wildcard'ing mx, and backupmx
   protocol=dyndns2,                                         \\
-  login=my-dyndns.org-login,                                \\
-  password=my-dyndns.org-password,                          \\
+  login=my-dyndns.com-login,                                \\
+  password=my-dyndns.com-password,                          \\
   mx=a.host.willing.to.mx.for.me,backupmx=yes,wildcard=yes  \\
-  myhost.dyndns.org,my2ndhost.dyndns.org
+  myhost.dyndns.com,my2ndhost.dyndns.com

   ## multiple host update to the custom DNS service
   protocol=dyndns2,                                         \\
-  login=my-dyndns.org-login,                                \\
-  password=my-dyndns.org-password                           \\
+  login=my-dyndns.com-login,                                \\
+  password=my-dyndns.com-password                           \\
   my-toplevel-domain.com,my-other-domain.com
 EoEXAMPLE
 }
@@ -2234,8 +2234,8 @@
        '!active'  => 'The hostname specified is in a Custom DNS domain which has not yet been activated.',
        'abuse',   => 'The hostname specified is blocked for abuse; fill in the form at http://support.dyndns.org/abuse.php to be unblocked',

-       'numhost'  => 'System error: Too many or too few hosts found. Contact support@dyndns.org',
-       'dnserr'   => 'System error: DNS error encountered. Contact support@dyndns.org',
+       'numhost'  => 'System error: Too many or too few hosts found. Contact support@dyndns.com',
+       'dnserr'   => 'System error: DNS error encountered. Contact support@dyndns.com',

        'nochg'    => 'No update required; unnecessary attempts to change to the current address are considered abusive',
     );
@@ -2507,7 +2507,7 @@

 Configuration variables applicable to the 'hammernode1' protocol are:
   protocol=hammernode1         ##
-  server=fqdn.of.service       ## defaults to members.dyndns.org
+  server=fqdn.of.service       ## defaults to members.dyndns.com
   login=service-login          ## login name and password  registered with the service
   password=service-password    ##
   fully.qualified.host         ## the host registered with the service.
